assignment3/src/pages/mutual-funds/search.tsx

```diff
@@ -0,0 +1,182 @@
+      //console.log(res);
+      setResults(res);
+    } catch (e) {
+      console.error("Failed to search for mutual funds", e);
+      setResults([]);
+    } finally {
+      setIsSearching(false);
+    }
+  };
+
+  const getSearchResults = async (searchWord: string) => {
+    console.log("searchWord: " + searchWord);
+    const resp = await axios.get("https://api.mfapi.in/mf/search", {
+      params: {
+        q: searchWord,
+      },
+    });
+
+    const json = await resp.data;
+
+    if (json.length === 0) {
+      setEmptyResult("No Results!")
+    }
+    console.log(json);
+
+    let resultSet: SearchResult[] = [];
+
+    json.forEach((searchLine : SearchResponse) => {
+      let searchResult: SearchResult = {
+        code: searchLine.schemeCode,
+        name: searchLine.schemeName,
+      };
+      resultSet.push(searchResult);
+    });
+    console.log(resultSet);
+
+/*    
+    // TODO: parse this from the `resp`
+    const res = [
+      {
+        code: "100522",
+        name: "Franklin India Technology Fund-Growth",
+      },
+      {
+        code: "101160",
+        name: "SBI MAGNUM DEBT FUND SERIES - 60 DAYS - 2 - GROWTH",
+      },
+      {
+        code: "119778",
+        name: "Kotak Global Emerging Market Fund - Payout of Income Distribution cum capital withdrawal option - Direct",
+      },
+    ];
+  */
+    return resultSet;
+  }
+
+  return (
+    <Box width="100%">
+      <Grid container spacing={1}>
+        <Grid item xs={12}>
+          <TextField 
+            placeholder="Search..."
+            fullWidth 
+            onChange={e => setSearchWord(e.target.value)}
+            InputProps={{
+              endAdornment: (
+                <InputAdornment position="end">
+                  <IconButton>
+                  <Search />
+                  </IconButton>
+                </InputAdornment>
+              ),
+            }}
+          />
+        </Grid>
+
+        <Grid item xs={6}>
+          <Button
+            size="large"
+            fullWidth
+            variant="contained"
+            onClick={() => searchResults(searchWord)}
+          >
+            Search
+          </Button>
+        </Grid>
+
+        <Grid item xs={6}>
+          <Button 
+            size="large" 
+            fullWidth 
+            variant="contained"
+            onClick={() => openFirstResult(searchWord)}>
+            I'm feeling lucky
+          </Button>
+        </Grid>
+      </Grid>
+
+      <Box mt={4} textAlign="center">
+        {isSearching ? (
+          <CircularProgress size="10vw" />
+        ) : results.length === 0 ? (
+          <Typography>{emptyResult}</Typography>
+        ) : (
+          <Box textAlign="end">
+            <Typography>Found {results.length} results</Typography>
+            <List>
+              {results.map((res) => (
+                <ListItem key={res.code}>
+                  <Link component={RouterLink} to={res.code} onClick = {() => {
```

Comment:
Using both `to` and `onClick` will not work as expected. Remove the `onClick` and pass the correct value to `to` instead. Check the value&type and make sure it's a string that points to the relative path from current page. See [docs](https://reactrouter.com/en/main/components/link) for help